protocol-units/mempool/util/src/lib.rs

```diff
@@ -1,172 +1,224 @@
 use serde::{Deserialize, Serialize};
 
-use movement_types::{Block, Transaction, Id};
+use movement_types::{Block, Id, Transaction};
 use std::cmp::Ordering;
+use thiserror::Error;
+
+pub type BoxedStoreError = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+#[derive(Error, Debug)]
```

Comment:
It's probably also good to annotate the error enums with `#[non_exhaustive]` so that adding new variants would not break consumer code.

Reply:
Good point, added